### ISSUE TITLE
📊 FEATURE: Add advanced analytics module with statistical testing and ML integration

### ISSUE THREAD

**Issue by mihiarc** [mihiarc/socialmapper#72]
### Objective
Enhance SocialMapper with advanced analytical capabilities for research and policy analysis.

### Feature 1: Statistical Significance Testing

```python
# socialmapper/analytics/statistics.py
import scipy.stats as stats
import numpy as np
from typing import Dict, List, Tuple

class AccessibilityStatistics:
    """Statistical analysis for accessibility metrics."""
    
    @staticmethod
    def equity_gap_test(
        group1_access: List[float],
        group2_access: List[float],
        test_type: str = 'mann-whitney'
    ) -> Dict:
        """
        Test for significant differences in accessibility between groups.
        
        Parameters
        ----------
        group1_access : List[float]
            Accessibility scores for group 1
        group2_access : List[float]
            Accessibility scores for group 2
        test_type : str
            Statistical test to use ('t-test', 'mann-whitney', 'ks')
        
        Returns
        -------
        Dict
            Test statistic, p-value, and interpretation
        """
        if test_type == 't-test':
            statistic, pvalue = stats.ttest_ind(group1_access, group2_access)
        elif test_type == 'mann-whitney':
            statistic, pvalue = stats.mannwhitneyu(group1_access, group2_access)
        elif test_type == 'ks':
            statistic, pvalue = stats.ks_2samp(group1_access, group2_access)
        
        return {
            'statistic': statistic,
            'p_value': pvalue,
            'significant': pvalue < 0.05,
            'interpretation': self._interpret_results(pvalue, test_type)
        }
    
    @staticmethod
    def spatial_autocorrelation(
        values: np.ndarray,
        weights_matrix: np.ndarray
    ) -> Dict:
        """Calculate Moran's I for spatial autocorrelation."""
        from pysal.explore import esda
        
        moran = esda.Moran(values, weights_matrix)
        return {
            'morans_i': moran.I,
            'p_value': moran.p_norm,
            'z_score': moran.z_norm,
            'clustered': moran.I > 0 and moran.p_norm < 0.05
        }
```

### Feature 2: Accessibility Scoring Algorithms

```python
# socialmapper/analytics/scoring.py
class AccessibilityScorer:
    """Calculate standardized accessibility scores."""
    
    @staticmethod
    def two_step_fca(
        population: pd.DataFrame,
        facilities: pd.DataFrame,
        travel_times: np.ndarray,
        threshold: int = 30
    ) -> pd.Series:
        """
        Two-Step Floating Catchment Area method.
        
        Enhanced gravity model for accessibility measurement.
        """
        # Step 1: Calculate facility-to-population ratios
        facility_ratios = {}
        for idx, facility in facilities.iterrows():
            catchment_pop = population[
                travel_times[idx] <= threshold
            ]['population'].sum()
            
            if catchment_pop > 0:
                facility_ratios[idx] = facility['capacity'] / catchment_pop
        
        # Step 2: Sum ratios for each population location
        accessibility_scores = pd.Series(index=population.index, dtype=float)
        for pop_idx in population.index:
            accessible_facilities = [
                f_idx for f_idx in facility_ratios
                if travel_times[f_idx][pop_idx] <= threshold
            ]
            accessibility_scores[pop_idx] = sum(
                facility_ratios[f_idx] for f_idx in accessible_facilities
            )
        
        return accessibility_scores
    
    @staticmethod
    def hansen_accessibility(
        opportunities: pd.DataFrame,
        travel_costs: np.ndarray,
        beta: float = 0.1
    ) -> np.ndarray:
        """
        Hansen accessibility measure with distance decay.
        
        A_i = Σ(O_j * exp(-β * C_ij))
        """
        return np.sum(
            opportunities.values[:, np.newaxis] * np.exp(-beta * travel_costs),
            axis=0
        )
```

### Feature 3: Machine Learning Integration

```python
# socialmapper/analytics/ml_models.py
from sklearn.ensemble import RandomForestRegressor
from sklearn.model_selection import cross_val_score
import joblib

class AccessibilityPredictor:
    """ML models for accessibility prediction."""
    
    def __init__(self):
        self.model = RandomForestRegressor(
            n_estimators=100,
            max_depth=10,
            random_state=42
        )
        self.feature_names = None
    
    def train(
        self,
        features: pd.DataFrame,
        accessibility_scores: pd.Series
    ):
        """
        Train model to predict accessibility from demographic features.
        
        Features might include:
        - Population density
        - Income levels
        - Age distribution
        - Transportation infrastructure
        """
        self.feature_names = features.columns.tolist()
        self.model.fit(features, accessibility_scores)
        
        # Cross-validation
        scores = cross_val_score(
            self.model, features, accessibility_scores,
            cv=5, scoring='r2'
        )
        
        return {
            'r2_scores': scores,
            'mean_r2': scores.mean(),
            'feature_importance': dict(zip(
                self.feature_names,
                self.model.feature_importances_
            ))
        }
    
    def predict_accessibility(
        self,
        new_features: pd.DataFrame
    ) -> pd.Series:
        """Predict accessibility for new areas."""
        predictions = self.model.predict(new_features)
        return pd.Series(predictions, index=new_features.index)
    
    def identify_underserved(
        self,
        features: pd.DataFrame,
        threshold_percentile: int = 25
    ) -> pd.DataFrame:
        """Identify potentially underserved areas."""
        predictions = self.predict_accessibility(features)
        threshold = np.percentile(predictions, threshold_percentile)
        
        underserved = features[predictions < threshold].copy()
        underserved['predicted_accessibility'] = predictions[predictions < threshold]
        underserved['accessibility_gap'] = threshold - underserved['predicted_accessibility']
        
        return underserved.sort_values('accessibility_gap', ascending=False)
```

### Feature 4: Time Series Analysis

```python
# socialmapper/analytics/temporal.py
class TemporalAccessibility:
    """Analyze accessibility changes over time."""
    
    @staticmethod
    def trend_analysis(
        yearly_data: Dict[int, pd.DataFrame]
    ) -> Dict:
        """Analyze trends in accessibility over years."""
        from scipy import stats
        
        years = sorted(yearly_data.keys())
        mean_scores = [yearly_data[year]['accessibility'].mean() 
                      for year in years]
        
        # Linear regression for trend
        slope, intercept, r_value, p_value, std_err = stats.linregress(
            years, mean_scores
        )
        
        return {
            'trend_direction': 'improving' if slope > 0 else 'declining',
            'annual_change': slope,
            'r_squared': r_value ** 2,
            'significant_trend': p_value < 0.05,
            'forecast_next_year': intercept + slope * (max(years) + 1)
        }
```

### Usage Examples

```python
from socialmapper.api import get_census_data, create_isochrone
from socialmapper.analytics import AccessibilityStatistics, AccessibilityScorer

# Analyze equity gaps
low_income_access = calculate_access_scores(low_income_areas)
high_income_access = calculate_access_scores(high_income_areas)

stats = AccessibilityStatistics()
equity_test = stats.equity_gap_test(
    low_income_access,
    high_income_access,
    test_type='mann-whitney'
)

print(f"Significant equity gap: {equity_test['significant']}")
print(f"P-value: {equity_test['p_value']:.4f}")

# Calculate accessibility scores
scorer = AccessibilityScorer()
scores = scorer.two_step_fca(
    population_data,
    facility_data,
    travel_time_matrix,
    threshold=20
)

# Identify underserved areas
predictor = AccessibilityPredictor()
predictor.train(demographic_features, scores)
underserved = predictor.identify_underserved(new_area_features)
```

### Benefits
- Enable academic research with statistical rigor
- Support evidence-based policy decisions
- Identify equity gaps quantitatively
- Predict future accessibility needs

**Labels:** enhancement, feature, analytics
